### PR TITLE
Issue20 wide2long

### PR DESCRIPTION
**Pull request by hswerdfe**
SQL matches the meta data better.
SQL is now for SQLite DB
Some R functions for working with the data an transforming from wide to long format

### CODE REVIEW

File: src/wbe_db_func.r [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+}
+
+#wbe_db_create_db_from_script()
+#wbe_db_delete_whole_DB()
+
+
+
+
+
+wbe_db_xlsx_to_dfs <- function(full_fn,
```
Comment (DougManuel) [inline src/wbe_db_func.r:113]: We haven't talked about data type. I was actually thinking we should stick to multiple csv files as opposed to xlsx for the usual reasons of challenges with binary files and proprietary software. 

Or at least default to csv format.

File: src/wbe_db_func.r [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+  tbl_format <- wbe_db_tbl_frmt(tbl_nm = tbl_nm, conn = conn)
+  for (col_nm in colnames(tbl_format)){
+    
+    curr_case <- ""
+    if( col_nm == "tableName"){curr_case <- table_name_cases}
+    else if(col_nm == "columnName"){curr_case <- col_name_cases}
+    else if(col_nm == "value"){curr_case <- value_name_cases}
+    else{curr_case <- desc_name_case}
+    #correct casing in the DB
+    df2[[col_nm]] <- df[[col_nm]] %>% snakecase::to_any_case(case = curr_case)
```
Comment (DougManuel) [inline src/wbe_db_func.r:169]: Is there a missing argument.

**Review by DougManuel — APPROVED** [review]
I didn't review in detail. I did execute the code and there was no errors. 

Using SQL in this code looks good. I was originally envisioning a more lightweight approach (like dyplr). 

Great to have all this work done. These helper functions are very, 'er, helpful and good to include in the repo. 

I am comfortable with R being the default language.

Next step, I think, is Rmd vignette, but at some point, it may make sense to have a separate repo that becomes a package with functions to use the data. With this repo a public aggregator of actual data.

File: src/wbe_db_func.r (hunk 1 above)
Comment (hswerdfe) [reply src/wbe_db_func.r:113]: good point, there is nothing wrong with providing multiple methods, and is xlsx is easy to work with out of the box and for lab techs to enter into, and switching to and from a well formated xlsx to multiple csv should be trivial.

**Pull request merged**